Add a label to the URL field in the Publishing Flow panel

afercia opened the pull request:
## Description

The Post URL field in the Publishing Flow panel is unlabelled. This PR adds a visible label. Also simplifies a bit the text for the Scheduled posts. I know there are pending proposal to improve and add features to the publishing flow so this part will be very likely improved and iterated. The most important thing in this PR is the input field label.

## Screenshots <!-- if applicable -->

Before:

<img width="700" alt="screen shot 2018-09-01 at 14 16 53" src="https://user-images.githubusercontent.com/1682452/44947053-6fe3bc00-ae07-11e8-9829-cffb4eea9035.png">

After:

<img width="700" alt="screen shot 2018-09-01 at 15 46 11" src="https://user-images.githubusercontent.com/1682452/44947056-78d48d80-ae07-11e8-8040-7e4b8e710943.png">

Note 1:
the sentence "The post address will be:" is not appropriate anyways, as a post can be of different types: post, page, or a custom post type. To keep things simple I've just removed it in favor of "What’s next?".

Note 2:
since the URL field is `readonly`, I'd suggest to communicate to users it can't be edited. Usually in WordPress this is done by using a light gray background. Something like in the screenshot below or a lighter gray could work:

<img width="281" alt="screen shot 2018-09-01 at 16 29 05" src="https://user-images.githubusercontent.com/1682452/44947071-b3d6c100-ae07-11e8-9912-6b07a1eeb689.png">


Fixes #9535

afercia committed:
Add a visible label.

jorgefilipecosta commented on packages/editor/src/components/post-publish-panel/postpublish.js:
```diff
@@ -48,24 +49,32 @@ class PostPublishPanelPostpublish extends Component {
+					<p className="post-publish-panel__postpublish-subheader">
+						<strong>{ __( 'What’s next?' ) }</strong>
+					</p>
+					<label htmlFor={ id } className="post-publish-panel__postpublish-link-label">
```
Can we use [Text Control ](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/text-control) component to simplify this logic? (it automatically generates the ids).

jorgefilipecosta reviewed (commented):
Hi @afercia thank you for looking into this issue. I did some tests, and things worked correctly 👍  From the code point view the changes look good, as a possible improvement we may check if Text Control component can be used here to simplify things and promote reusability.

afercia committed:
Use TextControl to simplify.

afercia commented:
Hi @jorgefilipecosta, thanks. Updated to use `TextControl`. I've also added a light grey background to the readonly field. I've opted for `$light-gray-400` `#e8eaeb` because it's the closest one to the grey used in core (`#eeeeee`).

One thing I don't like so much in the `TextControl` component is that the label element, by default, is styled as `display: block`. Thus, all the available width is clickable:

<img width="310" alt="textcontrol label" src="https://user-images.githubusercontent.com/1682452/45031900-84ff5b80-b050-11e8-951f-93951e77b365.png">

This could be confusing for users, especially when the clickable area is very large. Should go in a separate issue though.

Note:
I see a few warnings (not recognized props on DOM elements) when the Publish panel opens, seem unrelated to me and they happen also on master.

karmatosed commented:
Looks like there are some conflicts on this. I am approving design wise but we would need to get those resolve, thanks!

jorgefilipecosta reviewed (approved):
I rebased this PR (resolving the conflict) so the local tests take into account changes occurred in parallel.
The tests went well and the code looks good to me. Thank you for iterating on this changes 👍

afercia commented:
Thank you Jorge.

chrisvanpatten commented:
@karmatosed I also wonder if it's even better to switch over to "Permalink" since that's a standard and familiar WP term already used in a few places.
